### PR TITLE
Installs a handler to delete active pods on termination

### LINKED ISSUE #41: Clean up pods when killed
Related to #38, but deleting a Calrissian job will not automatically delete the pods it submitted to the API.

We should clean these up when the process is killed. Suggestion: catch SIGTERM and delete any submitted pods.

### PR DESCRIPTION
**Pull request by dleehr**
- Adds `k8s.PodMonitor` class that keeps track of submitted but not deleted pods
- Adds `k8s.delete_pods()` function to be called on sigterm or if `cwlmain()` raises an exception
- installs `SIGTERM` handler into `main.main()` to invoke `k8s.delete_pods()`
- Addresses a couple inconsistencies in tests

Fixes #41

### CODE REVIEW

File: calrissian/k8s.py [hunk 1]
```diff
@@ -46,6 +47,7 @@ def __init__(self):
     def submit_pod(self, pod_body):
         pod = self.core_api_instance.create_namespaced_pod(self.namespace, pod_body)
         log.info('Created k8s pod name {} with id {}'.format(pod.metadata.name, pod.metadata.uid))
+        PodMonitor.add(pod)
```
Comment (johnbradley) [inline calrissian/k8s.py]: Is there a possibility that a SIGTERM arrives just before this method, the main thread runs `delete_pods()` and the pod created here is leaked?

File: tests/test_k8s.py [hunk 2]
```diff
@@ -98,7 +97,8 @@ def test_wait_skips_pod_when_state_is_running(self, mock_watch, mock_get_namespa
         self.assertIsNotNone(kc.pod)
 
     @patch('calrissian.k8s.watch')
-    def test_wait_finishes_when_pod_state_is_terminated(self, mock_watch, mock_get_namespace, mock_client):
+    @patch('calrissian.k8s.PodMonitor')
+    def test_wait_finishes_when_pod_state_is_terminated(self, mock_podmonitor, mock_watch, mock_get_namespace, mock_client):
```
Comment (johnbradley) [inline tests/test_k8s.py]: Shouldn't we test that PodMonitor.remove is called here?

File: calrissian/k8s.py (hunk 1 above)
Comment (johnbradley) [reply calrissian/k8s.py]: Shouldn't we test that `PodMonitor.add` is called when users run `submit_pod`

**Comment by dleehr** [comment]
Thanks for the feedback @johnbradley. I made a few changes to guard against the case you described and added some documentation on threading assumptions.

**Pull request merged**